plugins/net_plugin/net_plugin.cpp

```diff
@@ -1195,6 +1206,27 @@ namespace eosio {
         last_handshake_recv(),
         last_handshake_sent()
    {
+      std::istringstream in(limit_str);
+      fc_dlog( logger, "parsing connection endpoint with locale ${l}", ("l", std::locale("").name()));
+      in.imbue(std::locale(""));
+      double limit{0};
+      in >> limit;
+      if( limit > 0.0f ) {
+         std::string units;
+         in >> units;
+         std::regex units_regex{"([KMGT]?[i]?)B/s"};
+         std::smatch units_match;
+         std::regex_match(units, units_match, units_regex);
+         if( units_match.size() == 2 ) {
+            block_sync_rate_limit = static_cast<size_t>(limit * prefix_multipliers.at(units_match[1].str()));
+            fc_dlog( logger, "setting block_sync_rate_limit to ${limit}", ("limit", block_sync_rate_limit));
+         } else {
+            fc_wlog( logger, "listen address ${la} has invalid block sync limit specification, connection will not be throttled", ("la", listen_address));
+            block_sync_rate_limit = 0;
+         }
+      } else {
+         block_sync_rate_limit = static_cast<size_t>(limit);
```

Comment:
This does not seem what you would want if limit is < 0.

Reply:
Negative limits are not supported and make no sense at all.

Reply:
I have no idea why I didn't just hard code the limit to 0 in that case, as I do on a failed parse.  Changed.